docs/source/build_python_model.rst

```diff
@@ -1,4 +1,104 @@
 Build a SMAUG model with Python API
 ===================================
 
-Building a model is quite easy in SMAUG.
+SMAUG's Python frontend provides easy APIs to build DL models. The created model
+is a computational graph, which are serialized into two protobuf files, one for
+the model topology and the other for the parameters. These two files are inputs
+to SMAUG's C++ runtime that performs the actual simulation. In this tutorial,
+we will be using the SMAUG Python APIs to build new DL models.
+
+Before building a model, we need to create a `Graph` context in which we will
+add operators.
+
+.. code-block:: python
+
+    import smaug as sg
+    with sg.Graph(name="my_model", backend="SMV") as graph:
+      # Any operators instantiated within the context will be added to `graph`.
+
+When using the :class:`smaug.Graph` API to create a graph context, we need to
+give a name for the model and select the backend we want to use when running the
+model through the C++ runtime. A backend in SMAUG is a logical combination of
+hardware blocks that implements all the SMAUG operators. Refer to
+`C++ docs <doxygen_html/index.html>`_ for more details of a backend
+implmentation in the C++ runtime. Here, we choose the `SMV` backend that comes
```

Comment:
It's worth mentioning the reference backend too.

Reply:
Right. Added reference backend too.